Scripts/DCS-BIOS/lib/modules/aircraft_modules/M-2000C.lua

```diff
@@ -149,6 +153,63 @@ local function build_pcn_segments(pcn, pcn_segment, display_name, length, includ
 	end
 end
 
+local function display_matrix_to_string(displayMatrix)
+	-- Mapping from binary segment pattern to character
+	local segment_to_char = {
+		[63] = "0",
+		[24] = "1",
+		[109] = "2",
+		[124] = "3",
+		[90] = "4",
+		[118] = "5",
+		[119] = "6",
+		[28] = "7",
+		[127] = "8",
+		[126] = "9",
+	}
+
+	local result = ""
+
+	for i = 1, #displayMatrix do
+		local segment = displayMatrix[i]
+
+		-- Check if all segments (except last) are off
+		local allOff = true
+		for j = 0, 6 do
+			if segment[j] and segment[j] > 0 then
+				allOff = false
+				break
+			end
+		end
+
+		if allOff and segment[7] and segment[7] > 0 then
+			result = result .. ". " -- Return decimal point and space if only decimal point is on
+		elseif allOff then
+			result = result .. " " -- Return space for all segments off
+		else
```

Comment:
I think you could simplify a lot of this conditional logic.

I suspect you could get rid of this, and then below after building `patternValue` do something like this (could maybe even be tightened further):
```lua
local result = ""
if segment[7] and segment[7] > 0 then
	result = result .. "."
end
if patternValue > 0 then
	result = result .. segment_to_char[patternValue] or "*"
else
	result = result .. " "
end
```
and then you could also remove your conditional segment 7 handling below as well

Reply:
Indeed, made it work first, didn't get back to make it in a smart way lol. Refactored it in a single line so was cleaner